x-pack/plugins/aiops/public/components/log_categorization/log_categorization_page.tsx

```diff
@@ -0,0 +1,327 @@
+}
+
+const BAR_TARGET = 20;
+
+export const LogCategorizationPage: FC<LogCategorizationPageProps> = ({
+  dataView,
+  savedSearch,
+}) => {
+  const {
+    notifications: { toasts },
+  } = useAiopsAppContext();
+
+  const { runCategorizeRequest, cancelRequest } = useCategorizeRequest();
+  const [aiopsListState, setAiopsListState] = useState(restorableDefaults);
+  const [globalState, setGlobalState] = useUrlState('_g');
+  const [selectedField, setSelectedField] = useState<string | undefined>();
+  const [selectedCategory, setSelectedCategory] = useState<Category | null>(null);
+  const [categories, setCategories] = useState<Category[] | null>(null);
+  const [currentSavedSearch, setCurrentSavedSearch] = useState(savedSearch);
+  const [loading, setLoading] = useState(false);
+  const [totalCount, setTotalCount] = useState(0);
+  const [eventRate, setEventRate] = useState<EventRate>([]);
+  const [pinnedCategory, setPinnedCategory] = useState<Category | null>(null);
+  const [sparkLines, setSparkLines] = useState<SparkLinesPerCategory>({});
+
+  useEffect(() => {
+    return () => {
+      cancelRequest();
+    };
+  }, [cancelRequest]);
+
+  const setSearchParams = useCallback(
+    (searchParams: {
+      searchQuery: Query['query'];
+      searchString: Query['query'];
+      queryLanguage: SearchQueryLanguage;
+      filters: Filter[];
+    }) => {
+      // When the user loads saved search and then clear or modify the query
+      // we should remove the saved search and replace it with the index pattern id
+      if (currentSavedSearch !== null) {
+        setCurrentSavedSearch(null);
+      }
+
+      setAiopsListState({
+        ...aiopsListState,
+        searchQuery: searchParams.searchQuery,
+        searchString: searchParams.searchString,
+        searchQueryLanguage: searchParams.queryLanguage,
+        filters: searchParams.filters,
+      });
+    },
+    [currentSavedSearch, aiopsListState, setAiopsListState]
+  );
+
+  const {
+    documentStats,
+    timefilter,
+    earliest,
+    latest,
+    searchQueryLanguage,
+    searchString,
+    searchQuery,
+    intervalMs,
+  } = useData(
+    { currentDataView: dataView, currentSavedSearch },
+    aiopsListState,
+    setGlobalState,
+    undefined,
+    BAR_TARGET
+  );
+
+  useEffect(() => {
+    if (globalState?.time !== undefined) {
+      timefilter.setTime({
+        from: globalState.time.from,
+        to: globalState.time.to,
+      });
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [JSON.stringify(globalState?.time), timefilter]);
+
+  const fields = useMemo(
+    () =>
+      dataView.fields
+        .filter(
+          ({ displayName, esTypes, count }) =>
+            esTypes && esTypes.includes('text') && !['_id', '_index'].includes(displayName)
+        )
+        .map(({ displayName }) => ({
+          label: displayName,
+        })),
+    [dataView]
+  );
+
+  useEffect(() => {
```

Comment:
nit
```suggestion
  useEffect(function setSingleFieldAsSelected() {
```

Reply:
Updated in https://github.com/elastic/kibana/pull/139005/commits/1cbe46049fb0cd7711c670cabb0530474081e174